src/mesh/rbf.f90

```diff
@@ -0,0 +1,597 @@
+    END DO
+    ! Take linear polynomial into account
+    xTmp = xTmp + RBFRHS(nBP+1,:) ! constant part
+    xTmp(1) = xTmp(1) + DOT_PRODUCT(RBFRHS(nBP+2:nBP+4,1),x(:)) ! Linear part
+    xTmp(2) = xTmp(2) + DOT_PRODUCT(RBFRHS(nBP+2:nBP+4,2),x(:)) ! Linear part
+    xTmp(3) = xTmp(3) + DOT_PRODUCT(RBFRHS(nBP+2:nBP+4,3),x(:)) ! Linear part
+    ! Overwrite curved node position by interpolation
+    Elem%CurvedNode(i)%np%x = xTmp
+  END DO
+  Elem=>Elem%nextElem
+END DO
+
+DEALLOCATE(RBFRHS)
+WRITE(UNIT_stdOut,'(A)')' VOLUME CURVING DONE!'
+
+END SUBROUTINE RBFVolumeCurving
+
+
+
+FUNCTION EvaluateRBF(dist,supportRadius,RBFType)
+!===================================================================================================================================
+! Evaluates the choosen RBF at distance dist with support radius supportRadius
+! Numbering and choice of RBFs is taken from: A. de Boer et al. / Computers and Structures 85 (2007) 784–795
+! The multiquadratic biharmonic RBFs (type 10 and 11) take an additional parameter (a) which controls the shape of the function.
+! Typical values for this parameter are in the range 10E-5 - 10E-3.
+!===================================================================================================================================
+USE MOD_Globals
+! INPUT VARIABLES
+REAL,INTENT(IN)    :: dist          ! Euclidean distance used in RBF evaluation
+REAL,INTENT(IN)    :: supportRadius ! Normalizing constant for RBF evaluation
+INTEGER,INTENT(IN) :: RBFType       ! Type of RBF to evaluate
+!-----------------------------------------------------------------------------------------------------------------------------------
+! OUTPUT VARIABLES
+REAL            :: EvaluateRBF
+!-----------------------------------------------------------------------------------------------------------------------------------
+! LOCAL VARIABLES
+REAL             :: xi
+REAL,PARAMETER   :: a=10E-3
+!===================================================================================================================================
+! normalized distance
+xi = dist/supportRadius
+
+SELECT CASE(RBFType)
+! Local support RBFs
+CASE(1)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**2.
+  END IF
+CASE(2)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF=(1.-xi)**4.*(4.*xi+1.)
+  END IF
+CASE(3)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**6.*((35./3.)*(xi**2.)+6.*xi+1.)
+  END IF
+CASE(4)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**8.*(32.*(xi**3.)+25.*(xi**2.)+8.*xi+1.)
+  END IF
+CASE(5)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**5.
+  END IF
+CASE(6)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = 1.+(80./3.)*(xi**2.)-40.*(xi**3.)+15.*(xi**4.)-(8./3.)*(xi**5.)+20.*(xi**2.)*LOG(xi)
+  END IF
+  IF (xi.LT.10E-13) EvaluateRBF=1.
```

Comment:
Maybe this should scale with `SpaceQuandt` for very small/large meshes?

Reply:
The supportRadius should be scaled accordingly.